### PR TITLE
Build: make sure to test all callback args

### PR DESCRIPTION
**Pull request by erikkemperman**
Playing around with an idea (not sure yet if I want to submit that) I found it convenient to always assert the outputs of all the streams, this helps narrow things down quickly when tests fail.

Note: If you guys agree with this principle, I guess the open PRs https://github.com/gulpjs/gulp-cli/pull/75 and https://github.com/gulpjs/gulp-cli/pull/113 might have to be amended accordingly. Let me know if I can help out.

### CODE REVIEW

**Comment by sttk** [comment]
@erikkemperman Thank you for pointing out. I agree and I’ll amend the added test codes in #75 and #113 according to the principle.

**Comment by erikkemperman** [comment]
@sttk It seems that I inadvertedly managed to decrease the coverage here and there, but I haven't figured out yet how exactly. Should I worry about that, do you think?

File: test/completion.js [hunk 1]
```diff
@@ -31,8 +33,9 @@ describe('flag: --completion', function() {
       .gulp('--completion=unknown')
       .run(cb);
 
-    function cb(err, stdout) {
-      expect(err).toExist();
+    function cb(err, stdout, stderr) {
+      expect(err).toNotEqual(null);
```
Comment (phated) [inline test/completion.js]: I prefer `toExist`, why did you change to `toNotEqual(null)`?

File: test/completion.js [hunk 2]
```diff
@@ -44,7 +47,9 @@ describe('flag: --completion', function() {
       .run(cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toNotEqual(null);
```
Comment (phated) [inline test/completion.js]: I prefer `toExist`, why did you change to `toNotEqual(null)`?

File: test/config-description.js [hunk 3]
```diff
@@ -20,7 +20,9 @@ describe('config: description', function() {
       .gulp('--tasks')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/config-description.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-description.js [hunk 4]
```diff
@@ -36,7 +38,9 @@ describe('config: description', function() {
...
```
Comment (phated) [inline test/config-description.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-description.js [hunk 5]
```diff
@@ -52,7 +56,9 @@ describe('config: description', function() {
       .gulp('--tasks', '--gulpfile ../foo/bar/gulpfile.js', '--cwd .')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/config-description.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-flags-gulpfile.js [hunk 6]
```diff
@@ -17,14 +17,15 @@ describe('config: flags.gulpfile', function() {
       .run(cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/config-flags-gulpfile.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-flags-gulpfile.js [hunk 7]
```diff
@@ -36,14 +37,15 @@ describe('config: flags.gulpfile', function() {
...
```
Comment (phated) [inline test/config-flags-gulpfile.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-flags-gulpfile.js [hunk 8]
```diff
@@ -56,12 +58,13 @@ describe('config: flags.gulpfile', function() {
...
```
Comment (phated) [inline test/config-flags-gulpfile.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-flags-gulpfile.js [hunk 9]
```diff
@@ -74,12 +77,13 @@ describe('config: flags.gulpfile', function() {
...
```
Comment (phated) [inline test/config-flags-gulpfile.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-flags-silent.js [hunk 10]
```diff
@@ -19,8 +19,9 @@ describe('config: flags.silent', function() {
       .run(cb);
 
     function cb(err, stdout, stderr) {
-      expect(stdout).toEqual('');
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/config-flags-silent.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/config-flags-silent.js [hunk 11]
```diff
@@ -33,13 +34,14 @@ describe('config: flags.silent', function() {
       .run(cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/config-flags-silent.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/exports-as-tasks.js [hunk 12]
```diff
@@ -19,9 +19,10 @@ describe('exports as tasks', function() {
         '--gulpfile ./test/fixtures/gulpfiles/gulpfile-exports.babel.js')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
```
Comment (phated) [inline test/exports-as-tasks.js]: `err` not tested?

File: test/flags-gulpfile.js [hunk 13]
```diff
@@ -17,7 +17,10 @@ describe('flag: --gulpfile', function() {
       .gulp('--gulpfile', gulpfilePath)
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-gulpfile.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-help.js [hunk 14]
```diff
@@ -21,7 +21,9 @@ describe('flag: --help', function() {
       .gulp('--help', '--cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-help.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-help.js [hunk 15]
```diff
@@ -33,7 +35,9 @@ describe('flag: --help', function() {
       .gulp('--h', '--cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-help.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-require.js [hunk 16]
```diff
@@ -15,7 +15,9 @@ describe('flag: --require', function() {
       .gulp('--require ../test-module.js', '--cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-require.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-require.js [hunk 17]
```diff
@@ -54,6 +56,9 @@ describe('flag: --require', function() {
       .run(cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-require.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-silent.js [hunk 18]
```diff
@@ -10,7 +10,9 @@ describe('flag: --silent', function() {
       .gulp('--silent', '--cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-silent.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-tasks-json.js [hunk 19]
```diff
@@ -15,7 +15,9 @@ describe('flag: --tasks-json', function() {
       .gulp('--tasks-json --gulpfile ./test/fixtures/gulpfiles/gulpfile.js')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-tasks-json.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-tasks-json.js [hunk 20]
```diff
@@ -33,7 +35,11 @@ describe('flag: --tasks-json', function() {
           '--gulpfile ./test/fixtures/gulpfiles/gulpfile.js')
         .run(cb);
 
-      function cb(err) {
+      function cb(err, stdout, stderr) {
+        expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-tasks-json.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-tasks-simple.js [hunk 21]
```diff
@@ -15,7 +15,9 @@ describe('flag: --tasks-simple', function() {
       .gulp('--tasks-simple', '--cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-tasks-simple.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-tasks.js [hunk 22]
```diff
@@ -16,7 +16,9 @@ describe('flag: --tasks', function() {
       .gulp('--tasks --cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-tasks.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-tasks.js [hunk 23]
```diff
@@ -32,7 +34,9 @@ describe('flag: --tasks', function() {
         '--cwd ./test/fixtures')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-tasks.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-tasks.js [hunk 24]
```diff
@@ -49,7 +53,9 @@ describe('flag: --tasks', function() {
...
```
Comment (phated) [inline test/flags-tasks.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-verify.js [hunk 25]
```diff
@@ -33,7 +33,9 @@ describe('flag: --verify', function() {
       .gulp('--verify valid-package.json', '--cwd ./test/fixtures/packages/')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/flags-verify.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/flags-version.js [hunk 26]
```diff
@@ -14,7 +14,8 @@ describe('flag: --version', function() {
       .gulp('--version --cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
-    function cb(err, stdout) {
+    function cb(err, stdout, stderr) {
```
Comment (phated) [inline test/flags-version.js]: `err` still not tested?

File: test/logging.js [hunk 27]
```diff
@@ -9,11 +9,12 @@ describe('logging', function() {
     child.exec('node ' + __dirname + '/fixtures/logging.js -LLLL', cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/logging.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/logging.js [hunk 28]
```diff
@@ -22,10 +23,11 @@ describe('logging', function() {
     child.exec('node ' + __dirname + '/fixtures/logging.js', cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/logging.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/logging.js [hunk 29]
```diff
@@ -34,10 +36,11 @@ describe('logging', function() {
     child.exec('node ' + __dirname + '/fixtures/logging.js -LLL', cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/logging.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/logging.js [hunk 30]
```diff
@@ -46,9 +49,10 @@ describe('logging', function() {
     child.exec('node ' + __dirname + '/fixtures/logging.js -LL', cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/logging.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

File: test/logging.js [hunk 31]
```diff
@@ -57,7 +61,9 @@ describe('logging', function() {
     child.exec('node ' + __dirname + '/fixtures/logging.js -L', cb);
 
     function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
```
Comment (phated) [inline test/logging.js]: I prefer `toNotExist`, why did you choose to use `toEqual(null)`?

**Comment by erikkemperman** [comment]
Yeah, I just used the established pattern. You're right though, but perhaps we could address that separately? Otherwise this will have to wait until I get back from holiday in about 2 weeks.

**Comment by phated** [comment]
@erikkemperman Yeah, I think that's fair.

**Comment by phated** [comment]
Merged as https://github.com/gulpjs/gulp-cli/commit/0abb70d50433a861b47557ed78dadf32e7070377 (cherry-pick to drop the appveyor changes)
